### PR TITLE
sriov: raise proper exception when SR-IOV parameter is not supported

### PR DESCRIPTION
**Pull request by ffmancera**
Some SR-IOV parameters could be not supported by the device driver.
Nmstate must return a proper error message instead of a generic one.

```
libnmstate.error.NmstateLibnmError: Activate profile: eno1 failed:
reason=The device=eno1 does not support one or more of the SR-IOV parameters set.
```

Ref: https://bugzilla.redhat.com/1819588

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

**Comment by ffmancera** [comment]
This bug was already fixed but https://github.com/nmstate/nmstate/commit/52020431da53fa49ca6c1ffed135dc8af1b85fde#diff-29bb9e68a40715683d91daa004b56cce brought it back.

File: libnmstate/nm/connection.py [hunk 1]
```diff
@@ -324,6 +324,18 @@ def _activation_progress_check(self, action):
...
+            not self._is_activating()
+            and self._is_sriov_parameter_not_supported_by_driver()
+        ):
+            reason = (
+                f"The device={self.devname} does not support one or "
+                "more of the SR-IOV parameters set."
+            )
+            self._activation_clean_up()
+            self._ctx.fail(
+                NmstateLibnmError(f"{action} failed: reason={reason}")
```
Comment (cathay4t) [inline libnmstate/nm/connection.py:337]: Should be `NmstateValueError` as desire state is invalid for this SRIOV NIC.

File: libnmstate/nm/connection.py [hunk 2]
```diff
@@ -333,6 +345,15 @@ def _activation_progress_check(self, action):
                 NmstateLibnmError(f"{action} failed: reason={reason}")
             )
 
+    def _is_sriov_parameter_not_supported_by_driver(self):
+        if self.nmdevice:
+            return (
+                self.nmdevice.props.state_reason
+                == NM.DeviceStateReason.SRIOV_CONFIGURATION_FAILED
```
Comment (cathay4t) [inline libnmstate/nm/connection.py:352]: `return self.nmdevice and self.nmdevice.props.state_reason == NM.DeviceStateReason.SRIOV_CONFIGURATION_FAILED`

Comment (ffmancera) [reply libnmstate/nm/connection.py]: Done, thanks!

File: libnmstate/nm/connection.py (hunk 1 above)
Comment (ffmancera) [reply libnmstate/nm/connection.py]: Done, thanks!

**Pull request merged**